src/Infrastructure.EntityFramework/Vault/Repositories/CipherRepository.cs

```diff
@@ -447,32 +449,45 @@ public async Task ReplaceAsync(CipherDetails cipher)
                     {
                         var favorites = CoreHelpers.LoadClassFromJsonData<Dictionary<Guid, bool>>(cipher.Favorites);
                         favorites.Remove(cipher.UserId.Value);
-                        cipher.Favorites = JsonConvert.SerializeObject(favorites);
+                        cipher.Favorites = JsonSerializer.Serialize(favorites);
                     }
                 }
                 if (cipher.FolderId.HasValue)
                 {
                     if (cipher.Folders == null)
                     {
-                        cipher.Folders = $"{{{userIdKey}:\"{cipher.FolderId}\"}}";
+                        var jsonObject = new JsonObject(new[]
+                        {
+                            new KeyValuePair<string, JsonNode>(cipher.UserId.Value.ToString(), cipher.FolderId),
+                        });
+                        cipher.Folders = JsonSerializer.Serialize(jsonObject);
                     }
                     else
                     {
                         var folders = CoreHelpers.LoadClassFromJsonData<Dictionary<Guid, Guid>>(cipher.Folders);
                         folders.Add(cipher.UserId.Value, cipher.FolderId.Value);
-                        cipher.Folders = JsonConvert.SerializeObject(folders);
+                        cipher.Folders = JsonSerializer.Serialize(folders);
                     }
                 }
                 else
                 {
                     if (cipher.Folders != null && cipher.Folders.Contains(cipher.UserId.Value.ToString()))
                     {
-                        var folders = CoreHelpers.LoadClassFromJsonData<Dictionary<Guid, bool>>(cipher.Favorites);
+                        var folders = CoreHelpers.LoadClassFromJsonData<Dictionary<Guid, Guid>>(cipher.Folders);
                         folders.Remove(cipher.UserId.Value);
-                        cipher.Favorites = JsonConvert.SerializeObject(folders);
+                        cipher.Favorites = JsonSerializer.Serialize(folders);
```

Comment:
This section is seriously confusing. Are we working with folders or favorites?

I have a feeling 478 should be `cipher.Folders`

Reply:
It definitely should be, I partially fixed it by switching it to `.Folders` but forgot that last little bit. Technically isn't a part of the original issue but something I found along the way.